src/marqo/tensor_search/tensor_search.py

```diff
@@ -360,8 +360,14 @@ def add_documents(config: Config, add_docs_params: AddDocsParams):
     except errors.IndexNotFoundError:
         raise errors.IndexNotFoundError(f"Cannot add documents to non-existent index {add_docs_params.index_name}")
 
-    if len(add_docs_params.docs) == 0:
+    doc_count = len(add_docs_params.docs)
+
+    if doc_count == 0:
         raise errors.BadRequestError(message="Received empty add documents request")
+    elif doc_count > utils.read_env_vars_and_defaults_ints(EnvVars.MARQO_MAX_ADD_DOCS_COUNT):
+        raise errors.BadRequestError(message=f"Number of docs in add documents request ({doc_count}) exceeds limit of {utils.read_env_vars_and_defaults(EnvVars.MARQO_MAX_ADD_DOCS_COUNT)}. "
+                                     f"This limit can be increased by setting the environment variable `{EnvVars.MARQO_MAX_ADD_DOCS_COUNT}`."
+                                     f"If using the Python client, use `client_batch_size`. See https://marqo.pages.dev/1.3.0/API-Reference/documents/#body for more details.")
```

Comment:
Let's move this logic either to a function in a validation file, or as a validator in the `AddDocsParams` class, to prevent the add_documents function from growing even larger.

Reply:
Got this! This being said, it would probably be a good idea to move the `if doc_count == 0` check there as well.

Reply:
I added the validation function in the AddDocsParams file as per the suggestion of @farshidz